TodoApp.ConsoleApp/Repositories/XMLRepository/TodoRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Xml.Linq;
+using TodoApp.ConsoleApp.Repositories.Models;
+using TodoApp.ConsoleApp.Repositories.Interfaces;
+
+namespace TodoApp.ConsoleApp.Repositories.XmlRepository
+{
+    public class TodoRepository : RepositoryBase<TodoModel, Guid>, ITodoRepository
+    {
+        public TodoRepository(string path) : base(path, "todos")
+        {
+
+        }
+
+        protected override string IdName => "Id";
+
+
+        protected override TodoModel ElementToEntity(XElement element)
+        {
+            if (element == null)
+            {
+                return null;
+            }
+
+            var entity = new TodoModel();
+            entity.Title = XmlParser.GetString(element, "Title");
+            entity.Description = XmlParser.GetString(element, "Description");
+            entity.CreatedOn = XmlParser.GetDateTime(element, "CreatedOn");
+            entity.DueDate = XmlParser.GetDateTime(element, "DueDate");
+            entity.Id = XmlParser.GetGuid(element, IdName);
+
+            return entity;
```

Comment:
We forgot the Status of the Todo. You have to Get and Set the todo.Status.

Create two new methods in `XmlParser`: 
- `GetEnum<T>`
- `SetEnum<T>`
They should be generic so they can be used for every enum (not only TodoStatus). Make sure you have added constraint so `T` can be only Enum.
You can use [Enum.Parse](https://docs.microsoft.com/en-us/dotnet/api/system.enum.parse?view=netframework-4.8) and [Enum.GetName](https://docs.microsoft.com/en-us/dotnet/api/system.enum.getname?view=netframework-4.8)